projects/addon-doc/src/components/language-switcher/language-switcher.component.ts

```diff
@@ -0,0 +1,70 @@
+import {
+    ApplicationRef,
+    ChangeDetectionStrategy,
+    ChangeDetectorRef,
+    Component,
+    Inject,
+    Input,
+} from '@angular/core';
+import {FormControl} from '@angular/forms';
+import {tuiDefaultProp} from '@taiga-ui/cdk';
+import {TuiDropdownWidthT, TuiSizeL, TuiSizeS} from '@taiga-ui/core';
+import {TuiLanguageName, TuiLanguageSwitcher} from '@taiga-ui/i18n';
+
+// @dynamic
+@Component({
+    selector: `tui-language-switcher`,
+    templateUrl: `./language-switcher.component.html`,
+    styleUrls: [`./language-switcher.component.less`],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class TuiLanguageSwitcherComponent {
+    @Input(`tuiTextfieldSize`)
+    size: TuiSizeS | TuiSizeL = `m`;
+
+    @Input(`tuiDropdownLimitWidth`)
+    @tuiDefaultProp()
+    limitWidth: TuiDropdownWidthT = `min`;
+
+    language = new FormControl(this.switcher.currentLanguage);
+
+    flags: Map<TuiLanguageName, string> = new Map([
+        [`chinese`, `🇨🇳 chinese`],
+        [`dutch`, `🇳🇱 dutch`],
+        [`english`, `🇬🇧 english`],
+        [`french`, `🇫🇷 french`],
+        [`german`, `🇩🇪 german`],
+        [`italian`, `🇩🇪 italian`],
+        [`polish`, `🇵🇱 polish`],
+        [`portuguese`, `🇵🇹 portuguese`],
+        [`russian`, `🇷🇺 russian`],
+        [`spanish`, `🇪🇸 spanish`],
+        [`turkish`, `🇹🇷 turkish`],
+        [`ukrainian`, `🇺🇦 ukrainian`],
+        [`vietnamese`, `🇻🇳 vietnamese`],
+    ]);
+
+    names: TuiLanguageName[] = Array.from(this.flags.keys());
+
+    constructor(
+        @Inject(TuiLanguageSwitcher) readonly switcher: TuiLanguageSwitcher,
+        @Inject(ChangeDetectorRef) private readonly cd: ChangeDetectorRef,
+        @Inject(ApplicationRef) private readonly app: ApplicationRef,
+    ) {}
+
+    set(lang: TuiLanguageName): void {
+        this.switcher.setLang(lang);
+        this.forceUpdateView();
+    }
+
+    reset(): void {
+        this.switcher.removeLang();
+        this.language.setValue(this.switcher.currentLanguage);
+        this.forceUpdateView();
+    }
+
+    forceUpdateView(): void {
```

Comment:
Why do we need that? Changing the language with select control already marks the view for checking and triggers application tick, since we do that with UI.

Reply:
fixed